File: source_funcs/go_transcoder/FIND_MAXIMUM_HEIGHT_PYRAMID_FROM_THE_GIVEN_ARRAY_OF_OBJECTS/FIND_MAXIMUM_HEIGHT_PYRAMID_FROM_THE_GIVEN_ARRAY_OF_OBJECTS.go
package main

import "sort"

func f_gold(boxes []int, n int) int {
	sort.Ints(boxes)
	var ans int = 1
	var prev_width int = boxes[0]
	var prev_count int = 1
	var curr_count int = 0
	var curr_width int = 0
	for i := int(1); i < n; i++ {
		curr_width += boxes[i]
		curr_count += 1
		if curr_width > prev_width && curr_count > prev_count {
			prev_width = curr_width
			prev_count = curr_count
			curr_count = 0
			curr_width = 0
			ans++
		}
	}
	return ans
}

File: rust_funcs/cpp_transcoder/FIND_MAXIMUM_HEIGHT_PYRAMID_FROM_THE_GIVEN_ARRAY_OF_OBJECTS/FIND_MAXIMUM_HEIGHT_PYRAMID_FROM_THE_GIVEN_ARRAY_OF_OBJECTS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::cmp::Ordering;

fn f_gold(mut boxes: [i32;2], n: i32) -> i32 {
    boxes.sort();
    let mut ans = 1;
    let mut prev_width = boxes[0]; 
    let mut prev_count = 1;
    let mut curr_count = 0;
    let mut curr_width = 0;
    for i in 1..n {
        curr_width += boxes[i as usize];
        curr_count += 1;
        if curr_width > prev_width && curr_count > prev_count {
            prev_width = curr_width;
            prev_count = curr_count;
            curr_count = 0;
            curr_width = 0;
            ans += 1;
        }
    }
    ans
}
